### PR TITLE
Refactors zookeeper dependencies and Kafka span receiver test

### PR DESCRIPTION
**Pull request by codefromthecrypt**
Zookeeper dependencies conflicted in most contexts. I set the minimum
version to coincide with the minimum used at Twitter internally, which
matches what Kakfa and Curator use.

Next, this ports the existing Kafka span receiver test to use a JUnit
rule, as opposed to manually orchestrating kafka and zookeeper. This
requires at least JUnit 4.11 for scala interop.

### CODE REVIEW

File: project/Project.scala [hunk 1]
```diff
@@ -379,9 +383,10 @@ object Zipkin extends Build {
     ).settings(
       libraryDependencies ++= Seq(
         twitterServer,
-        "org.apache.kafka" %% "kafka" % "0.8.1.1",
+        "org.apache.kafka" %% "kafka" % "0.8.2.1",
         "org.apache.commons" % "commons-io" % "1.3.2",
-        scroogeDep("serializer")
+        scroogeDep("serializer"),
+        "com.github.charithe" % "kafka-junit" % "1.3" % "test"
```
Comment (codefromthecrypt) [inline project/Project.scala]: @abesto everything above the kafka-junit line should be done to the build regardless of my change

Comment (abesto) [inline project/Project.scala]: Gotcha. Ping me once this is merged and I'll take care of the conflict accordingly.

**Comment by codefromthecrypt** [comment]
rebased over latest decoupled build

**Comment by codefromthecrypt** [comment]
@jamescway PTAL

**Comment by eirslett** [comment]
You can use `@Rule` in Scala if you use JUnitSuite:
http://doc.scalatest.org/1.7/org/scalatest/junit/JUnitSuite.html

Then you won't need the painful `$.MODULE$` stuff that comes from bad Java-Scala-interop... (I'm working on making Span more Java-friendly!)

**Comment by codefromthecrypt** [comment]
Thx will change it!

**Comment by codefromthecrypt** [comment]
@eirslett ok.. was a little tricky, but yeah better than interop on the Span type for sure. PTAL!

File: zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala [hunk 2]
```diff
@@ -1,47 +1,29 @@
...
 
-import com.twitter.zipkin.collector.{SpanReceiver, ZipkinQueuedCollectorFactory}
-import java.io._
+object KafkaProcessorSpecSimple {
+  val kafkaRule = new KafkaJunitRule()
 
-import com.twitter.finagle.stats.{DefaultStatsReceiver, StatsReceiver}
-import com.twitter.server.TwitterServer
-import com.twitter.zipkin.zookeeper.ZooKeeperClientFactory
+  @ClassRule def kafkaRuleDef = kafkaRule
```
Comment (codefromthecrypt) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:17]: this dance is to ensure the rule is a singleton.

Comment (rtyler) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:17]: why not put that comment into the code itself for posterity :)

Comment (codefromthecrypt) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:17]: roger that!

Comment (codefromthecrypt) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:17]: done

**Comment by rtyler** [comment]
an timid :+1: 

I think @eirslett should sign off and merge this if he's good with it

File: zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala [hunk 3]
```diff
@@ -1,47 +1,31 @@
...
-import kafka.serializer.Decoder
-import kafka.server.KafkaServer
+import org.junit.{ClassRule, Test}
+import org.scalatest.junit.JUnitSuite
 
-import com.twitter.zipkin.collector.{SpanReceiver, ZipkinQueuedCollectorFactory}
-import java.io._
+object KafkaProcessorSpecSimple {
+  // Singleton as the test needs to read the actual port in use
+  val kafkaRule = new KafkaJunitRule()
```
Comment (jamescway) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:14]: Pretty cool, [kafka-unit](https://github.com/chbatey/kafka-unit) allows you to delete all that test code

**Comment by jamescway** [comment]
:+1: :+1:  (u get 2)  super awesome!

**Comment by codefromthecrypt** [comment]
@eirslett forgive me for merging :) I think I addressed your concern and happy to revise further in a latter change as necessary!

File: zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala (hunk 3 above)
Comment (codefromthecrypt) [inline zipkin-receiver-kafka/src/test/scala/com/twitter/zipkin/receiver/kafka/KafkaProcessorSpec.scala:14]: thank @charithe!

**Comment by eirslett** [comment]
Great work! I love that you could delete all that code!
`kafkaRule` could maybe be a `lazy val`, but whatever.
Cheers! :beers:

**Pull request merged**